include/llvm/ADT/StringMap.h

```diff
@@ -382,7 +382,7 @@ class StringMap : public StringMapImpl {
         }
       }
     }
-    free(TheTable);
+    ::operator delete(TheTable); // HLSL Change Begin: Use overridable operator delete
```

Comment:
why is this qualification needed?

Reply:
The colon-colon? You're right, there's no reason, I can remove them

Reply:
Actually, if it was unqualified, it could pick up the custom operator delete of the class which defines the method (`StringMap::operator delete` in this case, if it was defined). So I think it's legit to be qualifying the calls to global operator new/delete.

Reply:
The .cpp has the matching `::operator new`